cmd/commands/addrs.go

```diff
@@ -82,7 +83,7 @@ func newAddr(ctx *cli.Context) error {
 	client, cleanUp := getClient(ctx)
 	defer cleanUp()
 
-	assetVersion, err := taprpc.MarshalAssetVersion(
+	assetVersion, err := rpcutils.MarshalAssetVersion(
```

Comment:
While this approach definitely works, I'm not sure if having a single `rpcutils` package is a very good idea. Because it will contain code that references many different packages, it will be very easy to create more circular dependencies later on.

What we instead should do in general, to avoid these "centralized" (from a dependency PoV) packages is to create a `marshal.go` file in each package and place these functions there.
And the functions would always go into the package that declares the non-RPC type of the marshal/unmarshal functions. So in this example it would be `asset`, because `asset.Version` is declared there.
Then we can also shorten the type a bit, as it can just be `asset.MarshalVersion`.

I know it's a bit more work but I'm sure we won't regret it in the future.

Reply:
One tradeoff with that approach is that then these packages that are mainly "pure" protocol biz logic (eg the group key verification logic in the `asset` package) will start to depend on the RPC sub-modules.

Reply:
Re circular deps, as long as we only import the package from the `main` package (rpc server logic, just encoding/decoding before we pass of into routines that live elsewhere), then I think we should be able to minimize the potential for circular deps. 

However if packages like say `rfq` do end up importing these new routines, then agree that it would be pretty easy to introduce one.

Reply:
Good point re depending on gRPC stuff from "pure" protocol logic packages. Which would make things hard if/when we decide to expose that logic as WASM ourselves.